src/readBig.ts

```diff
@@ -28,30 +26,41 @@ function boundsError(value: number, length: number) {
   throw ERR_OUT_OF_RANGE('offset', `>= 0 and <= ${length}`, value)
 }
 
-// https://github.com/nodejs/node/blob/v14.17.0/lib/internal/buffer.js#L129-L145
-export function readBigInt64LE(buffer: Buffer, offset = 0): bigint {
-  validateNumber(offset, 'offset')
-  const first = buffer[offset]
-  const last = buffer[offset + 7]
-  if (first === undefined || last === undefined) boundsError(offset, buffer.length - 8)
-  // tslint:disable-next-line:no-bitwise
-  const val = buffer[offset + 4] + buffer[offset + 5] * 2 ** 8 + buffer[offset + 6] * 2 ** 16 + (last << 24) // Overflow
-  return (
-    (BigInt(val) << BigInt(32)) + // tslint:disable-line:no-bitwise
-    BigInt(first + buffer[++offset] * 2 ** 8 + buffer[++offset] * 2 ** 16 + buffer[++offset] * 2 ** 24)
-  )
-}
-
-// https://github.com/nodejs/node/blob/v14.17.0/lib/internal/buffer.js#L89-L107
+// This function works with react-native >= 0.66.1
 export function readBigUInt64LE(buffer: Buffer, offset = 0): bigint {
-  validateNumber(offset, 'offset')
-  const first = buffer[offset]
-  const last = buffer[offset + 7]
-  if (first === undefined || last === undefined) boundsError(offset, buffer.length - 8)
+  buffer = buffer.slice(offset)
+
+  const bufferLenght = Math.min(buffer.length, 8)
+
+  let tot: number = 0
```

Comment:
I think this code loses precision -- this is a floating point number, which has fewer bits of precision than a 64 bit integer (since some of its 64 bits are allocated to the exponent). I think if you make this a BigInt right off the bat, it should be fine.